### ISSUE TITLE
Make Coq history publicly available

### ISSUE THREAD

**Issue by ppedrot** [coq/coq#6337]
An archive containing the source code of many old versions of Coq, including a RCS tree, has been circulating privately for a few years. I have already tried to press people so as to make it public, in order for the masses to dive into the delight of reading fancy mildly NSFW cursing comments in French from the end of the 80's, but the discussion has stalled several times.

As far as I can tell, this archive is still not publicly available.

A recent comment by @herbelin about history reminded me about this. I'm opening this issue for us not to forget about it.

**Comment by maximedenes** [comment]
I fully support this.

> I have already tried to press people so as to make it public

Can you say what is preventing it from happening? Licensing issues?

**Comment by ppedrot** [comment]
> Can you say what is preventing it from happening? Licensing issues?

Yes. The great Gérard was complaining that we shouldn't put a license on this code, while supporting public release at the same time. I think we can just dump the code somewhere though.

**Comment by Zimmi48** [comment]
It's perfectly OK to dump an archive with code and without a license. If the goal is really to allow public scrutiny, it is already good enough. Putting an open source license would be better, but it can come at a later time.

**Comment by SkySkimmer** [comment]
Isn't it illegal if you don't have a license?

**Comment by ejgallego** [comment]
@SkySkimmer not necessarily, it depends on who is the copyright holder.

**Comment by Zimmi48** [comment]
Well, sharing this code without the permission of the author / copyright holder is illegal but I don't think that is what is suggested here. Gérard, Coquand and Christine are probably OK to give permission to sharing this archive. Moreover the copyright holder is likely to be Inria anyways.

**Comment by maximedenes** [comment]
> Moreover the copyright holder is likely to be Inria anyways.

That part is far from clear. First, @mattam82 recently pointed me to an article explaining that researchers who are civil servants in France usually hold copyright on their work. This would have to be confirmed by a lawyer of course, but I did quite some research and was very surprised by what I read. The second problem is that there are many co-authors of Coq who were not working for Inria.

**Comment by Zimmi48** [comment]
OK, how many authors are there from these early days?

**Comment by herbelin** [comment]
> how many authors are there from these early days?

It depends. For versions 1 to 3: I guess only 3 or 4 authors. For versions 4.X: 5 or 6. For versions 5.X, maybe 10-12. Version 6.3.1 is listing 19 names. Looking at the dates in file `CREDITS` should give an indication of the number of contributors per year. When needed, we can find the exact figures.

The public archive includes versions from 7.0.

**Comment by ppedrot** [comment]
Judging from the lingering state of this issue, I solemnly declare hereby that I am going to dump those archives on my website, utterly disregarding the legal conundrum about this data. It will contain the note written by Gérard for attribution purposes.

I will calmly wait for any potential incoming takedown notice and subsequent judicial trouble. *Qu'ils viennent me chercher.*

**Comment by ejgallego** [comment]
@ppedrot why don't you discuss with the Software Heritage people first? They have the expertise to do this, let me know if I can help in some local way.

**Comment by ejgallego** [comment]
In particular a source code of the importance of coq could deserve hosting at the institutional level, some communication action, etc...

**Comment by Zimmi48** [comment]
I don't see why this would need to be done *first*. I support all steps toward making this history public, both short-term and long-term.

**Comment by ejgallego** [comment]
It wouldn't hurt to discuss with experts first indeed, IMVHO.

**Comment by ejgallego** [comment]
The main problems is as @ppedrot says, stalling; but if you folks think it is a good idea I can take care of setting up a comm channel with them [and ccing you] and we see. If it stalls we can just have Pierre-Marie dumping the code in the web.

**Comment by Blaisorblade** [comment]
Let me 👍  and thank @ppedrot's courage. The Coq history talk at the CoqPL workshop https://coq-workshop.gitlab.io/2020/#invited-talk would be a good excuse to revive this effort 😉
